main.c

```diff
@@ -44,11 +32,132 @@ void setButtonState(int button, bool pressed) {
   }
 }
 
+int gamepad_map[] = {SDLK_z, SDLK_a, SDLK_RSHIFT, SDLK_RETURN, // B Y Sel Sta
+                     SDLK_UP, SDLK_DOWN, SDLK_LEFT, SDLK_RIGHT,
+                     SDLK_x, SDLK_s, SDLK_d, SDLK_c}; // A X L R
+int joypad_map[] = {
+  SDL_CONTROLLER_BUTTON_A, // B
+  SDL_CONTROLLER_BUTTON_X, // Y
+  SDL_CONTROLLER_BUTTON_BACK,
+  SDL_CONTROLLER_BUTTON_START,
+  SDL_CONTROLLER_BUTTON_DPAD_UP,
+  SDL_CONTROLLER_BUTTON_DPAD_DOWN,
+  SDL_CONTROLLER_BUTTON_DPAD_LEFT,
+  SDL_CONTROLLER_BUTTON_DPAD_RIGHT,
+  SDL_CONTROLLER_BUTTON_B, // A
+  SDL_CONTROLLER_BUTTON_Y, // X
+  SDL_CONTROLLER_BUTTON_LEFTSHOULDER,
+  SDL_CONTROLLER_BUTTON_RIGHTSHOULDER,
+};
+
+static uint8_t* readFile(char* name, size_t* length) {
+  FILE* f = fopen(name, "rb");
+  if(f == NULL) {
+    return NULL;
+  }
+  fseek(f, 0, SEEK_END);
+  int size = ftell(f);
+  rewind(f);
+  uint8_t* buffer = (uint8_t *)malloc(size);
+  fread(buffer, size, 1, f);
+  fclose(f);
+  *length = size;
+  return buffer;
+}
+
+void parseKeyconf(char* buf) {
+  char* names[] = {

```

Comment:
I don't like this way of parsing yaml. Why pretend it's yaml when it's not...

Reply:
Good point. the file resembles an .ini file, would be easier to write and parse that way.

Reply:
I could rename it to .ini but it's also wont support ini features either

So what shall we keep ? txt ?